src/native/libs/System.IO.Compression.Native/extra_libs.cmake

```diff
@@ -13,7 +13,7 @@ macro(append_extra_compression_libs NativeLibsExtra)
       find_package(ZLIB REQUIRED)
       list(APPEND ZLIB_LIBRARIES m)
   else()
-    list(APPEND ZLIB_LIBRARIES zlib)
+    list(APPEND ZLIB_LIBRARIES $<IF:$<BOOL:CLR_CMAKE_USE_SYSTEM_ZLIB>,z,zlib>)
```

Comment:
We should be able to delete the ANDROID special-case above since it should be handled by this condition just fine now.

Reply:
Good point. Done. I also added a comment.